### PR TITLE
Added e2e tests for `page.tag.attached` webhook

### PR DESCRIPTION
**Pull request by leonyangela**
ref #15537

Got some code for us? Awesome 🎊!

Please include a description of your change & check your PR against this list, thanks!

- [x] There's a clear use-case for this code change, explained below
- [x] Commit message has a short title & references relevant issues
- [ ] The build will pass (run `yarn test:all` and `yarn lint`)

We appreciate your contribution!

Also, if you'd be interested in writing code like this for us more regularly, we're hiring:
https://careers.ghost.org

### CODE REVIEW

File: ghost/core/test/e2e-webhooks/pages.test.js [hunk 1]
```diff
@@ -31,7 +43,7 @@ const buildAuthorSnapshot = (roles = false) => {
     };
 
     if (roles) {
-        return {...authorSnapshot, roles: Array(1).fill(roleSnapshot)};
+        return { ...authorSnapshot, roles: Array(1).fill(roleSnapshot) };
```
Comment (naz) [inline ghost/core/test/e2e-webhooks/pages.test.js:46]: No need for extra spaces here 🙃

File: ghost/core/test/e2e-webhooks/pages.test.js [hunk 2]
```diff
@@ -56,13 +68,47 @@ const buildPageSnapshotWithTiers = ({
     };
 };
 
+const buildPageSnapshotWithTiersAndTags = ({
```
Comment (naz) [inline ghost/core/test/e2e-webhooks/pages.test.js:71]: Would it make sense to reuse the `buildPageSnapshotWithTiers` method here?

File: ghost/core/test/e2e-webhooks/pages.test.js [hunk 3]
```diff
@@ -23,6 +24,17 @@ const tierSnapshot = {
 
 const roleSnapshot = tierSnapshot;
 
+const tagSnapshot = {
```
Comment (naz) [inline ghost/core/test/e2e-webhooks/pages.test.js:27]: The tag's name, slug, visibility etc. are not changing between test runs? If they are not, would be best to have snapshot capture them as is - no need to include non-dynamic properties in the snapshot matcher object 👍

File: ghost/core/test/e2e-webhooks/pages.test.js [hunk 4]
```diff
@@ -171,4 +217,61 @@ describe('page.* events', function () {
...
+                    {
+                        title: 'test page.tag.attached webhook',
+                        status: 'draft',
+                        
+                    }
+                ]
+            })
+            .expectStatus(201);
+
+            console.log(res);
```
Comment (naz) [inline ghost/core/test/e2e-webhooks/pages.test.js:242]: Probably a leftover debugging code 😄

**Review by naz — CHANGES_REQUESTED** [review]
Hey @leonyangela! Thanks for taking time to make the contribution. It's in pretty good shape just needs to polish up a few rough edges. I've left some guiding comments in the PR for the improvements. Feel free to reach out if you need more guidance 😊

**Comment by leonyangela** [comment]
Hi @naz, Thanks for your feedback. I've made another changes. I've tried to run `yarn lint` but there is an error saying that:
```
Oops! Something went wrong! :(

ESLint: 8.25.0

No files matching the pattern "'core/server/**/*.js'" were found.
Please check for typing mistakes in the pattern.

error Command failed with exit code 2.
```
Do you have any idea for this? I also made a changes on tagSnapshot, please let me know if there's more feedback. Thank you.

File: ghost/core/test/e2e-webhooks/pages.test.js (hunk 4 above)
Comment (leonyangela) [reply ghost/core/test/e2e-webhooks/pages.test.js]: I apologize, my bad.

File: ghost/core/test/e2e-webhooks/pages.test.js (hunk 1 above)
Comment (leonyangela) [reply ghost/core/test/e2e-webhooks/pages.test.js]: I apologize, my bad. I've resolved this, thank you.

**Comment by leonyangela** [comment]
Hi @naz, I've updated the tagSnapshot and made some changes. Please let me know if there's any more feedback, cheers!

**Comment by naz** [comment]
Hey @leonyangela , just a heads up there are some linting errors that need to be fixed. The rest of the code is heading in a right direction 👍

**Comment by leonyangela** [comment]
Hi @naz, the lint test flow keeps failing, I have no idea on this, any suggestions?

**Review by naz — APPROVED** [review]
The linting passed! 🚀  You got there eventually @leonyangela , thanks for the effort and great contribution 💪

**Comment by leonyangela** [comment]
ah great, thank you @naz.

**Comment by ErisDS** [comment]
Hi @leonyangela  thanks so much for this PR. It has now been merged 🎉  FYI I had to rebase it as it conflicted with changes made to add the `page.tag.detached` tests.

I'm not sure if you found this through hacktoberfest, but I've added the `hacktoberfest-accepted` label to this PR to make sure it counts. We have plenty of open issues and ongoing projects for hacktoberfest and beyond. We would love to see you around again 👋

**Comment by leonyangela** [comment]
Hi @ErisDS, thank you. Yes, I found this from hacktoberfest, thanks for adding the `hacktoberfest-accepted` label, cheers :beers:.

**Pull request merged**